docs/adr/000x-modelling-history.md

```diff
@@ -0,0 +1,314 @@
+  accessible;
+  *because* this maintains options for TimelineEntry and provides rich
+  debugging information.
+- Avoid relying on the TimelineEntry model as the main answer to any
+  historical queries;
+  *because* this model is close to the user and is likely to be iterated. The
+  deeper the coupling to a document the more difficult this becomes.
+
+
+### Could
+
+- Link TimelineEntry to changes rather than storing the information twice;
+  *because* this would reduce concerns and opportunities for error.
+- Consider states to be something that may change in a background job;
+  *because* publishing can be a slow task that could timeout a single
+  HTTP request, considering this up-front will be less painful than bolting it
+  on.
+- Enable a future feature to enable reverting to different points in time;
+  *because* this could enable users to make minor changes to a live document
+  and then recover a draft.
+- Separate related data to different tables to reduce data usage;
+  *because* storing each change a user makes could be inefficient for data
+  usage.
+
+### Won’t
+
+- Consider the sharing of data between translations, as per
+  [ADR-3](0003-initial-domain-modelling.md#decision);
+  *because* there is insufficient product decisions around what translation
+  needs/expectations are.
+
+## Decision
+
+The proposed data model expands on the definitions of entities accordingly:
+
+**Document**: A record that represents all versions of a piece of content in a
+particular locale. It has many editions and many revisions. At any time it will
+have a current edition -  shown on Content Publisher index - and potentially a
+live edition which is currently on GOV.UK. The live and current edition can be
+the same. Document serves as a mutable entity that tracks data across all
+editions (such as first publishing date) and it is expected to be a joining
+point for document related data that is not associated with a particular
+edition.
+
+**Edition**: A numbered version of a document that has been, or is
+expected to be, published on GOV.UK. It is associated with a revision
+and a status. It is mutable and is expected as a place where any edition level
+database constraints can be placed.
+
+**Revision**: Represents the content of a document at a particular point
+in time. It has a number to indicate which revision of the document it is and
+stores who created it. Any time a user changes the data of a document a new
+revision object is created and the edition is updated to reference that
+revision. A revision itself cannot be edited and is considered immutable. A
+revision may have a association with a collection of items, such as image
+revisions, when a new revision is created new image revisions are not created
+but a new join between them is. A revision has a number of components which are
+described further in this document.
+
+This modelling now has a concept of:
+
+**Status**: Represents a particular singular state that an edition can
```

Comment:
Maybe lose the word 'singular' as I think its clearer to just put `Represents a particular state that an edition can hold ...`

Reply:
I went for the word "singular" to represent that it's meant to be just a single word that describes that edition as a whole. 

So if an edition is a "draft" it can't be "submitted for review", they're separate things. Whereas another aspect of state is `revision_synced` but that's not a singular thing as it can be applied to multiple statuses.

Though as I type this I don't think I'm doing too good a job of explaining here so it might be actually better to try work out how to rephrase it. I think what I'm trying to explain is that compared to other stateful aspects of an edition a status is special because it's 1) shown to the user 2) used for filtering and 3) can only be in one of them at a time.

Reply:
I've had a go at rephrasing this section now.